homeassistant/helpers/check_config.py

```diff
@@ -277,7 +277,7 @@ async def _get_integration(
                 try:
                     p_validated = platform_schema(p_validated)
                 except vol.Invalid as ex:
-                    _comp_error(ex, f"{domain}.{p_name}", p_validated)
+                    _comp_error(ex, f"{domain}.{p_name}", p_config)
```

Comment:
Does `_comp_error` handle a config that is not exactly the same as the one that failed to validate?

Reply:
Yes, `_comp_error` attaches the failing config item, allowing it to be printed by the `check_config` script. If we attach the validated config, it may be very different from what the user has in their configuration.

Reply:
Ok.